doc/source/security/using-kmip.rst

```diff
@@ -0,0 +1,98 @@
+
+==========================================================
+Using the Key Management Interoperability Protocol (KMIP)
+==========================================================
+
+This feature is **technical preview** quality.
+
+**Percona Server for MySQL** 8.0.27-18 adds support for Key Management Interoperability Protocol (KMIP).
+
+KMIP enables communication between key management systems and the database server. The protocol can do the following:
+
+* Streamline encryption key management
+
+* Eliminate redundant key management processes
+
+
+
+Component installation
+--------------------------------------
+
+Only one keyring plugin or one keyring component should be enabled at a time. Enabling multiple keyring plugins or keyring components are not supported and may result in data loss.
+
+.. note::
+
+    At one time, only one keyring component or plugin should be enabled. Multiple components or plugins being enabled at the same time is not supported and may have unexpected results.
+
+Install Component
+---------------------------
+
+The KMIP component must be installed. A keyring component is not loaded with the ``--early-plugin-load`` option on the server. The server uses a manifest and the component consults its configuration file during initialization. 
+
+To install a keyring component, you must do the following:
+
+1. Write a manifest in a valid JSON format
+2. Write a configuration file
+
+You should create a global manifest file named ``mysqld.my`` in the installation directory and, optionally, create a local manifest file, also named ``mysqld.my`` in the data directory.
+
+This is an example of a local configuration file:
+
+.. code-block:: json
+
+   {"server_addr": "127.0.0.1",
+   "server_port": "5696",
+   "client_ca": "client_certificate.pem",
+   "client_key": "client_key.pem",
+   "server_ca": "root_certificate.pem"
+   }
+
+To use a global configuration file and a local configuration file pair, a global file example would be:
+
+.. code-block:: json
+
+    {
+        "read_local_config": true
+    }
+
+The local configuration file example would have this information and created in the data directory:
+
+.. code-block:: json
+
+    {
+        "path":"usr/.local/etc/kmip_configuration_file",
+        "read-only": false
+    }
+
+For more information, see `Keyring Component installation <https://dev.mysql.com/doc/refman/8.0/en/keyring-component-installation.html>`__ 
+
+Usage
+-------------------------
+
+The certificates used by KMIP may either be self-signed using ``openssl`` or created by a certificate authority.
+
+The KMIP variables are the connection parameters: the host/port of the KMIP server and the certificates used for the SSL communication.
+
+The following is an example of the KMIP configuration:
+
+.. code-block:: text
+
+    [client]
+    host=(hostname)
+    port=5696
+    certfile=(client certificate)
+    keyfile=(client key)
+    ca_certs=(client certificate authority)
+    ssl_version=PROTOCOL_TLSv1_2
```

Comment:
This is a configuration for PyKMIP, it has nothing to do with mysql or this component. The component configuration is the JSON shown earlier.

Reply:
removed.